src/System.Device.Gpio/System/Device/Spi/SpiDevice.cs

```diff
@@ -7,13 +7,23 @@ namespace System.Device.Spi
     /// <summary>
     /// The communications channel to a device on a SPI bus.
     /// </summary>
-    public abstract partial class SpiDevice : IDisposable
+    public abstract class SpiDevice : IDisposable
     {
+        public SpiDevice(SpiConnectionSettings settings, Board board)
+        {
+            ConnectionSettings = settings;
```

Comment:
One thing that has worked well for Spi and I2c is the factory pattern for creating instances, I would encourage to keep using it specially since we have differnet platform implementations underneath.